index.bs

```diff
@@ -73,34 +73,125 @@ command/response format of WebDriver, this permits events to stream
 from the user agent to the controlling software, better matching the
 evented nature of the browser DOM.
 
+Infrastructure {#infrastructure}
+==========================
+
+This specification depends on the Infra Standard. [[!INFRA]]
+
+Network protocol messages are defined using CDDL. [[!RFC8610]]
+
 Protocol {#protocol}
 ==============
 
 This section defines the basic concepts of the WebDriver BiDi
 protocol. These terms are distinct from their representation at the
 <a href=#transport>transport</a> layer.
 
-This specification uses Concise Data Definition Language (CDDL)
-[[!RFC8610]] to describe the format of messages transmitted between the
-[=local end=] and [=remote end=].
+The protocol is defined using [[!RFC8610|CDDL]] schema. For the convenience of
+implementors two seperate schema are defined; the <dfn>remote end schema</dfn>
+which defines the format of messages produced on the [=local end=] and consumed
+on the [=remote end=], and the <dfn>local end schema</dfn> which defines the
+format of messages produced on the [=remote end=] and consumed on the [=local
+end=]
+
+## Schema ## {#protocol-schema}
+
+Issue: Should this be an appendix?
+
+This section gives the initial contents of the [=remote end schema=] and
+[=local end schema=]. These are augmented by the schema fragments defined in
+the remainder of the specification.
+
+[=Remote end schema=]
+
+```
+Command = {
+  id: uint,
+  CommandData,
+}
+
+CommandData = (
+  SessionData
+)
+
+EmptyParams = { *text }
+```
+
+[=Local end schema=]
+
+```
+Message = (
+  CommandResponse //
+  Event //
+)
 
-Note: These definitions do not form any normative requirements, rather
-they are used elsewhere as part of such requirements.
+CommandResponse = (
+  Error //
+  CommandResult
+)
+
+Error = (
+  id: unit,
```

Comment:
```suggestion
  id: uint,
```

Also wouldn't it make sense to move this into the `CommandResponse` group as we would always expect an `id` to a response, no?

Reply:
Good point.